boa/src/builtins/object/mod.rs

```diff
@@ -877,3 +913,48 @@ fn object_define_properties(
     // 7. Return O.
     Ok(())
 }
+
+/// Type enum used in the abstract operation GetOwnPropertyKeys
+#[derive(Debug, Copy, Clone)]
+enum GetOwnPropertyKeysType {
+    String,
+    Symbol,
+}
+
+/// The abstract operation GetOwnPropertyKeys
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-getownpropertykeys
+fn get_own_property_keys(
+    o: &JsValue,
+    r#type: GetOwnPropertyKeysType,
+    context: &mut Context,
+) -> JsResult<JsValue> {
+    // 1. Let obj be ? ToObject(o).
+    let obj = o.to_object(context)?;
+
+    // 2. Let keys be ? obj.[[OwnPropertyKeys]]().
+    let keys = obj.__own_property_keys__(context)?;
+
+    // 3. Let nameList be a new empty List.
+    // 4. For each element nextKey of keys, do
+    let name_list: Vec<_> = keys
+        .iter()
+        .filter_map(|next_key| {
+            // a. If Type(nextKey) is Symbol and type is symbol or Type(nextKey) is String and type is string, then
+            // i. Append nextKey as the last element of nameList.
+            use GetOwnPropertyKeysType as Type;
+            match (r#type, next_key) {
+                (Type::String, PropertyKey::String(key)) => Some(key.clone().into()),
+                (Type::String, PropertyKey::Index(index)) => Some(index.to_string().into()),
+                (Type::Symbol, PropertyKey::Symbol(symbol)) => Some(symbol.clone().into()),
+                _ => None,
+            }
+        })
+        .collect();
```

Comment:
You can skip `collect`ing into a `Vec`. `create_array_from_list` accepts any `IntoIterator` as an argument. This also avoids allocating a new `Vec`.

Reply:
Oh I see, thanks for pointing this out :)